### DISCUSSION TITLE
Simplify documentation with inheritance.

### DISCUSSION THREAD

**Discussion by hallambaker** [dotnet/csharplang#4456] [Language Ideas] [answered]
I like to document my code but I find C# makes me do a lot of work that is repetitive in ways that tend to lead to inconsistency and error.

Consider the case where I have classes A, B, C which each declare one method:

class A {
       /// <summary>
        /// Some documentation
        /// </summary>
        /// <param name="param1">A parameter</param>
   public virtual int Method (string param1) => 1
   }

class B : A{
       /// <summary>
        /// The exact same documentation repeated 
        /// </summary>
        /// <param name="param1">A parameter</param>
   public override int Method (string param1) => 1
   }

class C : A {
       /// <summary>
        /// A different method with a new documentation.
        /// </summary>
        /// <param name="param1">A parameter with the same purpose as before</param>
   public virtual int Method2 (string param1) => 1
   }

Wouldn't it be much nicer if classes inherited documentation from the parent class for overriden methods in the exact same way that they inherit code?

Further, Wouldn't it be a big code improvement win if I could assign a specific documentation line to a parameter name so that it is then reused throughout a project?

class B : A{
       /// <override/>
       /// <note>Some additional information specific to the overriden implementation.</note>
   public override int Method (string param1) => 1
   }

class C : A {
       /// <summary>
        /// A different method with a new documentation.
        /// </summary>
        /// <link name="param1"/>
   public virtual int Method2 (string param1) => 1
   }

This would seem to compliment the recent thrust of C# language extensions to enable repetitive coding to be eliminated.

Of course, Sandcastle, would also need to be updated to recognize this stuff...

**Comment by 333fred** [comment, marked as answer]
I think you're looking for https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/inheritdoc.

**Reply by hallambaker** [reply]
That should do it, hadn't noticed it in the 16.4 release notes...

**Reply by 333fred** [reply]
No worries, there's lots of features being added all the time.
